wfe/web-front-end_test.go

```diff
@@ -1387,6 +1388,157 @@ func TestVerifyPOSTUsesStoredKey(t *testing.T) {
 	test.AssertError(t, err, "No error returned when provided key differed from stored key.")
 }
 
+func TestRejectsNone(t *testing.T) {
+	wfe, _ := setupWFE(t)
+	_, _, _, err := wfe.verifyPOST(newRequestEvent(), makePostRequest(`
+		{
+			"header": {
+				"alg": "none",
+				"jwk": {
+					"kty": "RSA",
+					"n": "vrjT",
+					"e": "AQAB"
+				}
+			},
+			"payload": "aGkK",
+			"signature": ""
+		}
+	`), true, "foo")
+	if err == nil {
+		t.Fatalf("verifyPOST did not reject JWS with alg: 'none'")
+	}
+	if err.Error() != "algorithm 'none' in JWS header not acceptable" {
+		t.Fatalf("verifyPOST rejected JWS with alg: 'none', but for wrong reason: %s", err)
+	}
+}
+
+func TestRejectsHS256(t *testing.T) {
+	wfe, _ := setupWFE(t)
+	_, _, _, err := wfe.verifyPOST(newRequestEvent(), makePostRequest(`
+		{
+			"header": {
+				"alg": "HS256",
+				"jwk": {
+					"kty": "RSA",
+					"n": "vrjT",
+					"e": "AQAB"
+				}
+			},
+			"payload": "aGkK",
+			"signature": ""
+		}
+	`), true, "foo")
+	if err == nil {
+		t.Fatalf("verifyPOST did not reject JWS with alg: 'HS256'")
+	}
+	expected := "algorithm 'HS256' in JWS header not acceptable"
+	if err.Error() != expected {
+		t.Fatalf("verifyPOST rejected JWS with alg: 'none', but for wrong reason: got '%s', wanted %s", err, expected)
+	}
+}
+
+func TestCheckAlgorithmES(t *testing.T) {
+	stat, err := checkAlgorithm(&jose.JsonWebKey{
+		Algorithm: "ES256",
+		Key:       &ecdsa.PublicKey{},
+	}, &jose.JsonWebSignature{})
+	expected := "no signature algorithms suitable for given key type"
+	if err == nil || err.Error() != expected {
+		t.Errorf("ES256 key: Expected error '%s', got '%s'", expected, err)
+	}
+	if stat != "NoAlgorithmForKey" {
+		t.Errorf("ES256 key: Expected stat '%s', got '%s'", "NoAlgorithmForKey", stat)
+	}
+}
+
+func TestCheckAlgorithmHS(t *testing.T) {
+	stat, err := checkAlgorithm(&jose.JsonWebKey{
+		Algorithm: "HS256",
+	}, &jose.JsonWebSignature{})
+	expected := "no signature algorithms suitable for given key type"
+	if err == nil || err.Error() != expected {
+		t.Errorf("HS256 key: Expected error '%s', got '%s'", expected, err)
+	}
+	if stat != "NoAlgorithmForKey" {
+		t.Errorf("HS256 key: Expected stat '%s', got '%s'", "NoAlgorithmForKey", stat)
+	}
+}
+
+func TestCheckAlgorithmBadJWS(t *testing.T) {
```

Comment:
Combine these CheckAlgorithms into one that has a slice of type []foobar where foobar is

    type foobar struct {
      key *jose.JsonWebKey
      sig *jose.JsonWebSignature // This could just be `HeaderAlgorithm string`, maybe
      expectedStat string
      expectedError string // these will still work for the empty success case with a slight change to code
    }